x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -267,17 +338,26 @@ export function getWaterfall(
       waterfallItem.docType === 'transaction' &&
       waterfallItem.id === entryTransactionId
   );
-  const itemsById: IWaterfallIndex = indexBy(waterfallItems, 'id');
   const orderedItems = entryTransactionItem
     ? getOrderedWaterfallItems(childrenByParentId, entryTransactionItem)
     : [];
+
   const traceRoot = getTraceRoot(childrenByParentId);
   const duration = getDuration(orderedItems);
   const traceRootDuration = traceRoot && traceRoot.transaction.duration.us;
   const services = getServices(orderedItems);
-  const getTransactionById = createGetTransactionById(itemsById);
   const serviceColors = getServiceColors(services);
-  const entryTransaction = getTransactionById(entryTransactionId);
+
+  const agentMarks = getAgentMarks(entryTransaction);
+  orderedItems.push(...agentMarks);
```

Comment:
@sqren. And finally I get the agent marks from the entry transaction and push it into orderedItems.